include/zephyr/sys/sem.h

```diff
@@ -138,6 +138,66 @@ int sys_sem_take(struct sys_sem *sem, k_timeout_t timeout);
  */
 unsigned int sys_sem_count_get(struct sys_sem *sem);
 
+/**
+ * @brief Leaves a code block guarded with @ref SYS_SEM_LOCK after releasing the
+ * lock.
+ *
+ * See @ref SYS_SEM_LOCK for details.
+ */
+#define SYS_SEM_LOCK_BREAK continue
+
+/**
+ * @brief Guards a code block with the given sys_sem, automatically acquiring
+ * the semaphore before executing the code block. The semaphore will be
+ * released either when reaching the end of the code block or when leaving the
+ * block with @ref SYS_SEM_LOCK_BREAK.
+ *
+ * @details Example usage:
+ *
+ * @code{.c}
+ * SYS_SEM_LOCK(&sem) {
+ *
+ *   ...execute statements with the semaphore held...
+ *
+ *   if (some_condition) {
+ *     ...release the lock and leave the guarded section prematurely:
+ *     SYS_SEM_LOCK_BREAK;
+ *   }
+ *
+ *   ...execute statements with the lock held...
+ *
+ * }
+ * @endcode
+ *
+ * Behind the scenes this pattern expands to a for-loop whose body is executed
+ * exactly once:
+ *
+ * @code{.c}
+ * for (sys_sem_take(&sem); ...; sys_sem_take(&sem)) {
```

Comment:
The third param is probably meant to be sys_sem_give(...).

Reply:
Fixed!